lib/transaction/input/multisigscripthash.js

```diff
@@ -19,16 +19,16 @@ var TransactionSignature = require('../signature');
 /**
  * @constructor
  */
-function MultiSigScriptHashInput(input, pubkeys, threshold, signatures, nestedWitness) {
+function MultiSigScriptHashInput(input, pubkeys, threshold, signatures, nestedWitness, opts) {
   /* jshint maxstatements:20 */
   Input.apply(this, arguments);
   var self = this;
   pubkeys = pubkeys || input.publicKeys;
   threshold = threshold || input.threshold;
   signatures = signatures || input.signatures;
   this.nestedWitness = nestedWitness ? true : false;
-  this.publicKeys = _.sortBy(pubkeys, function(publicKey) { return publicKey.toString('hex'); });
-  this.redeemScript = Script.buildMultisigOut(this.publicKeys, threshold);
+  this.publicKeys = pubkeys;
```

Comment:
I think you have to check the noSort opt here as well (leaving in the public key sort) to not break existing code